src/setup.js

```diff
@@ -6,6 +6,37 @@ import settings from './settings'
  * @param {Object} customSettings - Custom settings
  * @returns {Object} new
  */
-export default function setup (customSettings) {
-  return Object.assign(settings, customSettings)
+
+/** Allowed API versions */
+const VERSIONS = [1, 2]
+/** Allowed API regions */
+const REGIONS = ['us', 'eu']
+/** Map of API URLs by version and region */
+const API_MAP = {
+  2: {
+    us: 'https://api.evrythng.io/v2',
+    eu: 'https://api.eu.evrythng.io/v2'
+  },
+  1: {
+    us: 'https://api.evrythng.com',
+    eu: 'https://api-eu.evrythng.com'
+  }
+}
+
+export default function setup (newSettings = {}) {
+  const { apiUrl, apiVersion = 2, region = 'us' } = newSettings
+
+  if (!VERSIONS.includes(apiVersion)) {
+    throw new Error(`Invalid apiVersion: ${apiVersion}. Choose from ${VERSIONS.join(', ')}`)
+  }
+  if (!REGIONS.includes(region)) {
+    throw new Error(`Invalid region: ${region}. Choose from ${REGIONS.join(', ')}`)
+  }
+
+  // Set the API URL and region
+  newSettings.apiUrl = apiUrl || API_MAP[apiVersion][region]
+  newSettings.apiVersion = apiVersion
```

Comment:
Since `apiVersion` and `region` are destructured from `newSettings`, they do not need to be reassigned. And if they did, `Object.assign` would do that.

Reply:
@C-D-Lewis We need it because of the default values we set, otherwise `Object.assign` will not receive the default values we've specified during destructuring

Reply:
Nice! I didn't spot that.